torchtext/datasets/iwslt.py

```diff
@@ -70,178 +99,55 @@ def _construct_filepaths(paths, src_filename, tgt_filename):
-        IWSLT16.TED.tst2012.ar-en.ar
-        IWSLT16.TED.tst2012.ar-en.en
-        IWSLT16.TED.tst2012.cs-en.cs
-        IWSLT16.TED.tst2012.cs-en.en
-        IWSLT16.TED.tst2012.de-en.de
-        IWSLT16.TED.tst2012.de-en.en
-        IWSLT16.TED.tst2012.en-ar.ar
-        IWSLT16.TED.tst2012.en-ar.en
-        IWSLT16.TED.tst2012.en-cs.cs
-        IWSLT16.TED.tst2012.en-cs.en
-        IWSLT16.TED.tst2012.en-de.de
-        IWSLT16.TED.tst2012.en-de.en
-        IWSLT16.TED.tst2012.en-fr.en
-        IWSLT16.TED.tst2012.en-fr.fr
-        IWSLT16.TED.tst2012.fr-en.en
-        IWSLT16.TED.tst2012.fr-en.fr
-        IWSLT16.TED.tst2013.ar-en.ar
-        IWSLT16.TED.tst2013.ar-en.en
-        IWSLT16.TED.tst2013.cs-en.cs
-        IWSLT16.TED.tst2013.cs-en.en
-        IWSLT16.TED.tst2013.de-en.de
-        IWSLT16.TED.tst2013.de-en.en
-        IWSLT16.TED.tst2013.en-ar.ar
-        IWSLT16.TED.tst2013.en-ar.en
-        IWSLT16.TED.tst2013.en-cs.cs
-        IWSLT16.TED.tst2013.en-cs.en
-        IWSLT16.TED.tst2013.en-de.de
-        IWSLT16.TED.tst2013.en-de.en
-        IWSLT16.TED.tst2013.en-fr.en
-        IWSLT16.TED.tst2013.en-fr.fr
-        IWSLT16.TED.tst2013.fr-en.en
-        IWSLT16.TED.tst2013.fr-en.fr
-        IWSLT16.TED.tst2014.ar-en.ar
-        IWSLT16.TED.tst2014.ar-en.en
-        IWSLT16.TED.tst2014.de-en.de
-        IWSLT16.TED.tst2014.de-en.en
-        IWSLT16.TED.tst2014.en-ar.ar
-        IWSLT16.TED.tst2014.en-ar.en
-        IWSLT16.TED.tst2014.en-de.de
-        IWSLT16.TED.tst2014.en-de.en
-        IWSLT16.TED.tst2014.en-fr.en
-        IWSLT16.TED.tst2014.en-fr.fr
-        IWSLT16.TED.tst2014.fr-en.en
-        IWSLT16.TED.tst2014.fr-en.fr
-        IWSLT16.TEDX.dev2012.de-en.de
-        IWSLT16.TEDX.dev2012.de-en.en
-        IWSLT16.TEDX.tst2013.de-en.de
-        IWSLT16.TEDX.tst2013.de-en.en
-        IWSLT16.TEDX.tst2014.de-en.de
-        IWSLT16.TEDX.tst2014.de-en.en
-        train.ar
-        train.ar-en.ar
-        train.ar-en.en
-        train.cs
-        train.cs-en.cs
-        train.cs-en.en
-        train.de
-        train.de-en.de
-        train.de-en.en
-        train.en
-        train.en-ar.ar
-        train.en-ar.en
-        train.en-cs.cs
-        train.en-cs.en
-        train.en-de.de
-        train.en-de.en
-        train.en-fr.en
-        train.en-fr.fr
-        train.fr
-        train.fr-en.en
-        train.fr-en.fr
-        train.tags.ar-en.ar
-        train.tags.ar-en.en
-        train.tags.cs-en.cs
-        train.tags.cs-en.en
-        train.tags.de-en.de
-        train.tags.de-en.en
-        train.tags.en-ar.ar
-        train.tags.en-ar.en
-        train.tags.en-cs.cs
-        train.tags.en-cs.en
-        train.tags.en-de.de
-        train.tags.en-de.en
-        train.tags.en-fr.en
-        train.tags.en-fr.fr
-        train.tags.fr-en.en
-        train.tags.fr-en.fr
+def IWSLT(root='.data', split=('train', 'valid', 'test'), offset=0, language_pair=('de', 'en'), year=2016, valid_set='dev2010', test_set='tst2010'):
```

Comment:
Let's split this into two datasets. One IWSLT2016 and one IWSLT2017. This should make the API easier.